EIPS/eip-4353.md

```diff
@@ -0,0 +1,230 @@
+const provider = waffle.provider;
+
+describe("StakedNFT", function () {
+    let _id = 1234567890;
+    let value = '1.5';
+    let Token;
+    let Interface;
+    let owner;
+    let addr1;
+    let addr2;
+
+    beforeEach(async function () {
+        Token = await ethers.getContractFactory("ERC721Staked");
+        [owner, addr1, ...addr2] = await ethers.getSigners();
+        Interface = await Token.deploy();
+    });
+
+    describe("Staked NFT", function () {
+        it("Should set the right owner", async function () {
+            let mint = await Interface.mint(
+                addr1.address, _id, 'http://foobar')
+            expect(await Interface.ownerOf(_id)).to.equal(addr1.address);
+        });
+
+        it("Should not have staked balance without value", async function () {
+            let mint = await Interface.mint(
+                addr1.address, _id, 'http://foobar')
+            expect(await Interface.stakedAmount(_id)).to.equal(
+                ethers.utils.parseEther('0'));
+        });
+
+        it("Should set and return the staked amount", async function () {
+            let mint = await Interface.mint(
+                addr1.address, _id, 'http://foobar',
+                {value: ethers.utils.parseEther(value)})
+            expect(await Interface.stakedAmount(_id)).to.equal(
+                ethers.utils.parseEther(value));
+        });
+
+        it("Should decrease owner eth balance on mint (deposit)", async function () {
+            let balance1 = await provider.getBalance(owner.address);
+            let mint = await Interface.mint(
+                addr1.address, _id, 'http://foobar',
+                {value: ethers.utils.parseEther(value)})
+            let balance2 = await provider.getBalance(owner.address);
+            let diff = parseFloat(ethers.utils.formatEther(
+                balance1.sub(balance2))).toFixed(1);
+            expect(diff === value);
+        });
+
+        it("Should add to payee's eth balance on burn (withdraw)", async function () {
+            let balance1 = await provider.getBalance(addr1.address);
+            let mint = await Interface.mint(
+                addr1.address, _id, 'http://foobar',
+                {value: ethers.utils.parseEther(value)})
+            await Interface.burn(_id);
+            let balance2 = await provider.getBalance(addr1.address);
+            let diff = parseFloat(ethers.utils.formatEther(
+                balance2.sub(balance1))).toFixed(1);
+            expect(diff === value);
+        });
+
+        it("Should update balance after transfer", async function () {
+            let mint = await Interface.mint(
+                addr1.address, _id, 'http://foobar',
+                {value: ethers.utils.parseEther(value)})
+            await Interface.burn(_id);
+            expect(await Interface.stakedAmount(_id)).to.equal(
+                ethers.utils.parseEther('0'));
+        });
+    });
+});
+```
+
+## Backward Compatibility
+TBD
```

Comment:
Not sure what happened to the other comment, but `Backwards Compatibility` has to come before `Rationale` to match the order in EIP-1.

Reply:
got it